utils.js

```diff
@@ -0,0 +1,68 @@
+// TODO
+// await use axios post seedData fn to push data to db
+// done: delete file directory(or add to gitignore so zip files doesn’t push to Github) + drain
+
+const fs = require('fs');
+const path = require('path');
+const unzipper = require('unzipper'); // npm lib used to unzip files
+const { createReadStream } = require('fs');
+
+// Async fs reads and unzip files to a directory
+const folderName = 'Archive.zip';
+const pathToExtract = 'unzippedArchive';
+
+const unzipFolder = (zippedFolder, destination) => {
+  fs.createReadStream(`./${zippedFolder}`).pipe(
+    unzipper.Extract({ path: __dirname + `/${destination}` })
+  );
+};
+
+class File {
+  constructor(filename) {
+    this.filename = filename;
+    this.fileContents;
+  }
+
+  // Read a file and save into the constructor
+  readFile() {
+    return new Promise((resolve, reject) => {
+      try {
+        fs.createReadStream(this.filename)
+
+          .on('data', chunk => {
+            this.fileContents = JSON.parse(chunk.toString());
+          })
+
+          .on('error', err => {
+            reject(err);
+          })
+
+          .on('close', () => {
+            resolve(this.fileContents);
+          });
+      } catch (err) {
+        reject();
+      }
+    });
+  }
+
+  // This method is here in the case we wanted to do invoke a function on the this.fileContents
+  async processFile(processFileFunction, ...otherArgs) {
```

Comment:
This method is unneeded, you're passing in a function and calling that function with the arguments you're passing in. Why not just do it inline?

Reply:
My thought process was that there would be different functions to format a file's parsed json data. When not also include a function that can be modular in such a way that one can pass in any type of function to perform and action... I.E., formatUsers(state.users), formatMessages(state.messages), formatChannels(state.channels). Does that make sense?

Reply:
So you want to defined functions like `formatUsers`, `formatMessages`, and `formatChannels` and pass that into `processFile` so it can call it for you? 

Example

```
const formatUser = (users) => {// Does something.};
const { users: usersState } = { state };
const file = new File('some path');

const usersResults = file(formatUser, usersState );
```

Also I just noticed you have `async` without a `await`.

Reply:
// Used to control the flow of reading a file.
const processFileController = async fileName => {
  let file = new File(fileName);
  try {
    // Read the file and store in the File object
    const results = await file.readFile();
    const filteredData = await state.readFile.processFile(func, results, funcArgs);
    return filteredData;
  } catch (err) {
    throw err;
  }
};

Reply:
`// Used to control the flow of reading a file.
const processFileController = async fileName => {
  let file = new File(fileName);
  try {
    // Read the file and store in the File object
    const results = await file.readFile();
    const filteredData = await state.readFile.processFile(func, results, funcArgs);
    return filteredData;
  } catch (err) {
    throw err;
  }
};`

Reply:
`

const processFileController = async fileName => {
  let file = new File(fileName);
  try {
  
    const results = await file.readFile();
  
  const filteredData = await state.readFile.processFile(func, results, funcArgs);
  
  return filteredData;
  } catch (err) {
    throw err;
  }
};

`

Reply:
And then invoke the processFileController in the switch statement like so. 

const users = await processFileController(absolutePath, filterUsers);